### PR TITLE
Identity Code Updates.

### LINKED ISSUE #19796: [BUG] Azure Identity - device code flow - wrong application
**Describe the bug**
Using the `DeviceCodeCredential` with Azure identity `1.2.4` the wrong application is presented once I enter the code.
After going to https://microsoft.com/devicelogin and entering the device code I get the following message for the account selection
You're signing in to **Microsoft Azure Cross-platform Command Line Interface** on another device located in Canada. If it's not you, close this page.

>Notice how the application name is the Azure CLI instead of being my application name.

If I go through the sign in, my application receives a token, however the appid is `04b07795-8ddb-461a-bbee-02f9e1bf7b46` instead of the one from my app registration.

***Exception or Stack Trace***
N/A

**To Reproduce**

1. create a console application
2. register a single tenant application that supports public clients
2. add azure-identity 1.2.4 as a dependency
3. in main, copy the following code sample (you can replace the tenant and client id by your own information)
4. run the application
5. (notice the application name in the sign in screens)
6. copy the token in jwt.ms and notice the appid is wrong

***Code Snippet***
```Java
final TokenCredential tokenCredential = new DeviceCodeCredentialBuilder()
                    .clientId("6755f70d-fac6-4b8b-a460-1510aaa95d1e")
                    .tenantId("bd4c6c31-c49c-4ab6-a0aa-742e07c20232")
                    .challengeConsumer(deviceCodeInfo -> {
                        System.out.println(deviceCodeInfo.getMessage());
                    }).build();
final String token = tokenCredential.getToken(new TokenRequestContext() {{
            this.setScopes(Arrays.asList(new String[] { "https://graph.microsoft.com/.default" }));
        }}).toFuture().thenApply(r -> r.getToken()).get();
System.out.println(token);
```

**Expected behavior**
The library to use the client id I'm providing.

**Screenshots**
image

**Setup (please complete the following information):**
 - OS: windows
 - IDE : VS Code
 - Version of the Library used 1.2.4

**Additional context**
I work on the Microsoft Graph Java SDKs.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes #19796 #18419  #19278 #18733

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredentialBuilder.java [hunk 1]
```diff
@@ -82,10 +81,7 @@ public DeviceCodeCredentialBuilder disableAutomaticAuthentication() {
      * @return a {@link DeviceCodeCredential} with the current configurations.
      */
     public DeviceCodeCredential build() {
-        ValidationUtil.validate(getClass().getSimpleName(), new HashMap<String, Object>() {{
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredentialBuilder.java:85]: These validations were focusing on the input being non-null.
Now the inputs have default values, so validation no longer needed.

File: sdk/identity/azure-identity/pom.xml [hunk 2]
```diff
@@ -98,6 +98,7 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- Optional: This dependency can be excluded if IntelliJ Credential is not being used for authentication via `IntelliJCredential` or `DefaultAzureCredential` -->
```
Comment (srnagar) [inline sdk/identity/azure-identity/pom.xml:101]: If a GA version of Identity is not shipped with `KeePassJava2` dependency in compile scope, we can set `<optional>true</optional>`

**Review by srnagar — APPROVED** [review]

File: sdk/identity/azure-identity/src/test/resources/settings.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    "window.zoomLevel": 0,
+    "java.errors.incompleteClasspath.severity": "ignore",
+    "workbench.startupEditor": "newUntitledFile",
+    "python.jediEnabled": false,
+    "java.configuration.checkProjectSettingsExclusions": false,
+    "java.requirements.JDK11Warning": false,
+    "java.semanticHighlighting.enabled": true,
+    "java.home": "/Contents/Home",
+    "python.languageServer": "Microsoft",
+  // this is a dummy comment.
```
Comment (jianghaolu) [inline sdk/identity/azure-identity/src/test/resources/settings.json:13]: JSON does not support comments - this may cause some unexpected behaviors?

**Review by jianghaolu — APPROVED** [review]

File: sdk/identity/azure-identity/src/test/resources/settings.json (hunk 3 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/src/test/resources/settings.json:13]: The azure maven plugin requested the feature to allow reading JSON files with comments in them for VS Code settings file.
So, the object mapper is configured to allow reading JSON files with comments in them.

**Pull request merged**